coreneuron/io/phase2.cpp

```diff
@@ -483,7 +483,7 @@ void Phase2::set_net_send_buffer(Memb_list** ml_list, const std::vector<int>& pn
             nrb->_pnt_offset = pnt_offset[type];
 
             // begin with a size of 5% of the number of instances
-            nrb->_size = ml->nodecount;
+            nrb->_size = 0.05 * ml->nodecount;
```

Comment:
I am not sure if the comment was outdated i.e. it should be `ml->nodecount` ? May be @nrnhines can confirm.

Reply:
The comment is outdated. The original was:
```
commit 921eabb6f65548d6c1fec8b6761d202d0a830abe (HEAD)
Author: Pramod Kumbhar <pramod.kumbhar@epfl.ch>
Date:   Wed May 25 16:25:38 2016 +0200

    NetSendBuffer_t is copied to GPU and updated back to CPU
    
    Change-Id: If0e3fe8ce74895f4378719465ef3686adcb5127c

diff --git a/coreneuron/nrniv/nrn_setup.cpp b/coreneuron/nrniv/nrn_setup.cpp
index 47c6c945..e61c36a5 100644
--- a/coreneuron/nrniv/nrn_setup.cpp
+++ b/coreneuron/nrniv/nrn_setup.cpp
@@ -1303,7 +1303,7 @@ for (int i=0; i < nt.end; ++i) {
       nrb->_pnt_offset = pnt_offset[type];
 
       // begin with a size of 5% of the number of instances
-      nrb->_size = ml->nodecount/20;
+      nrb->_size = ml->nodecount;
       // or at least 8
       if (nrb->_size < 8) {
         nrb->_size = 8;
```
I believe the earlier version was a space saving measure and referred to the presumption that incoming spikes are rare during a minimum delay interval. I'd replace the comment with
```
    // begin with a size equal to the number of instances
```
It would take a bit of thought to know if the size could, in reality , grow beyond that (e.g. many self events per instance) but there is, I believe, provision for growth in each translated mod file.

Reply:
Just to say that the reason this change is here is that I was debugging an issue that was triggered by the growth/reallocation. So: without this change then the reallocation was sometimes happening, and I originally made the change so that the bug was triggered earlier in execution.

I left the change in the PR in because of the inconsistency between code and comment, and wanting to trigger this discussion.

Reply:
> issue that was triggered by the growth/reallocation.

That is a bit disturbing.  Is growth itself problematic on the GPU or is it straightforward to fix the growth code? But the context of you seeing it in 2022 when the change was made in 2016 is confusing. I could write a nonsense mod file that generates several self events so that the growth code is triggered even though we are starting out with size nodecount.

Reply:
> nonsense mod file that generates several self events

Not exactly nonsense, but it is always better to minimize the size of the event priority queue by generating a single self event in response to the reception of the previous event rather that generating a flurry of self events all at once. In practice it is rare to have multiple self events outstanding for a given instance. Also, for a given instance only unique flag == 1 self events are movable to a new delivery time.

Reply:
Sorry, I mean that I (inadvertently) introduced a new bug "elsewhere" that happened to be triggered by the resizing code; to aid debugging that, I [re-]added the 5% so it crashed more quickly when I was debugging.

> I could write a nonsense mod file that generates several self events so that the growth code is triggered even though we are starting out with size nodecount.

I believe this already happens in the current test suite, in the `reduced_dentate` model, with the caveat:
https://github.com/neuronsimulator/nrn/blob/3cc83cf7b09044a478665490d52acfc31f606036/test/external/reduced_dentate/CMakeLists.txt#L12-L15

(although as I say, this was part of a debugging exercise, so this was an observation in a build that was known to have problems...)

Reply:
Ok. Understood.

Reply:
I updated this to initially allocate `ml->nodecount` entries again.